FirebaseRemoteConfig/Tests/Unit/RCNConfigContentTest.m

```diff
@@ -25,7 +25,36 @@
 #import "FirebaseRemoteConfig/Tests/Unit/RCNTestUtilities.h"
 
 @interface RCNConfigContent (Testing)
-- (void)checkAndWaitForInitialDatabaseLoad;
+- (BOOL)checkAndWaitForInitialDatabaseLoad;
+@end
+
+extern const NSTimeInterval kDatabaseLoadTimeoutSecs;
+@interface RCNConfigDBManagerMock : RCNConfigDBManager
+@property(nonatomic, assign) BOOL isLoadMainCompleted;
+@property(nonatomic, assign) BOOL isLoadPersonalizationCompleted;
+@end
+@implementation RCNConfigDBManagerMock
+- (void)createOrOpenDatabase {
+}
+- (void)loadMainWithBundleIdentifier:(NSString *)bundleIdentifier
+                   completionHandler:(RCNDBLoadCompletion)handler {
+  double justSmallDelay = 0.008;
+  XCTAssertTrue(justSmallDelay < kDatabaseLoadTimeoutSecs);
+  dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(justSmallDelay * NSEC_PER_SEC)),
+                 dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+                   self.isLoadMainCompleted = YES;
+                   handler(YES, nil, nil, nil);
+                 });
+}
+- (void)loadPersonalizationWithCompletionHandler:(RCNDBLoadCompletion)handler {
+  double justOtherSmallDelay = 0.009;
+  XCTAssertTrue(justOtherSmallDelay < kDatabaseLoadTimeoutSecs);
+  dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(justOtherSmallDelay * NSEC_PER_SEC)),
+                 dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+                   self.isLoadPersonalizationCompleted = YES;
+                   handler(YES, nil, nil, nil);
+                 });
+}
```

Comment:
I can see that OCMock is used for in other tests for mocking, but defining the mock as a separate class doesn't look like an issue to me in this case. @karenyz WDYT?

Reply:
AFAIK OCMock can not mock `- init`. But I wanted to override `- (void)createOrOpenDatabase` before `- init` fired to prevent heavy code of database loading.

Also in current way:
```
@property(nonatomic, assign) BOOL isLoadMainCompleted;
@property(nonatomic, assign) BOOL isLoadPersonalizationCompleted;
```
located in separate mock class, not in unit test class itself.

Reply:
With OCMock approach you probably would create a instance of a class mock with `OCMClassMock()`  macro and configure stubs and expectations before passing it to the `RCNConfigContent` initializer. There are examples of this approach in the repo e.g. a [relatively simple one](https://github.com/firebase/firebase-ios-sdk/blob/23b6acefb9afaf7dc4f5471a14fc087971878072/FirebaseRemoteConfig/Tests/Unit/RCNConfigAnalyticsTest.m#L32)  or more advanced like [here](https://github.com/firebase/firebase-ios-sdk/blob/23b6acefb9afaf7dc4f5471a14fc087971878072/FirebaseInstallations/Source/Tests/Unit/FIRInstallationsStoreTests.m#L140) or [here](https://github.com/firebase/firebase-ios-sdk/blob/23b6acefb9afaf7dc4f5471a14fc087971878072/FirebaseAuth/Tests/Unit/FIRPhoneAuthProviderTests.m#L537).

We usually prefer using OCMock to make our tests more uniform. I can see room for better test coverage for this class and adopting OCMock now should make this work easier in the future. On the other hand we have exceptions as well where manually created mocks/fakes are used, so I personally don't think it is a blocker for the current PR.